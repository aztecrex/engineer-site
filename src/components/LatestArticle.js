import React from 'react';
import R from 'ramda';
import {connect} from 'react-redux';
import Article from './Article';

class LatestArticle extends React.Component {

  render() {
    return <Article id={this.props.id} />
  }

}

const later = (e1,e2) => {
  return e2.published > e1.published ? e2 : e1;
};

const mapStateToProps = state => {
  const entries = R.path(['articles','directory'], state) || [];

  let id;
  if (entries.length == 0)
    id = null;
  else {
    id = R.reduce(later, R.head(entries), R.tail(entries)).id;
  }
  return {id};

};

export default connect(mapStateToProps)(LatestArticle);
export {LatestArticle, mapStateToProps};
