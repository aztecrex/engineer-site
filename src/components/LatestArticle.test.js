import React from 'react';
import configureMockStore from 'redux-mock-store';
const createMockStore = configureMockStore([]);
import ezJson from 'enzyme-to-json';

import {LatestArticle} from './LatestArticle';

describe('unconnected', () => {
  it ('renders', () => {

    // given
    const id = 'article883';

    // when
    const rendered = shallow(<LatestArticle id={id} />);

    // then
    expect(ezJson(rendered)).toMatchSnapshot();

  });
});
