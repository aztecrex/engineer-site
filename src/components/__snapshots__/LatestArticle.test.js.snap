exports[`connected connects state 1`] = `
<LatestArticle
  dispatch={[Function]}
  id="four"
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  } />
`;

exports[`unconnected renders 1`] = `
<Connect(Article)
  id="article883" />
`;
